File: 2444. Count Subarrays With Fixed Bounds/code.java
class Solution {
    public long countSubarrays(int[] nums, int minK, int maxK) {
        
    int n = nums.length;
    int startInd = 0;
    int lastMinInd = -1;
    int lastMaxInd = -1;
    long res = 0;
    for(int i = 0; i < n; i++){
        if(nums[i] < minK || nums[i] > maxK){
            lastMinInd = lastMaxInd = -1;
            startInd = i + 1;
            } 
            if(nums[i] == minK){
                lastMinInd = i;
                }
                if(nums[i] == maxK){
                    lastMaxInd = i;
                    }
                    res += Math.max(0L, Math.min(lastMinInd, lastMaxInd) - startInd + 1);
                    }
                    return res;
                    }
                }
    

File: 1456. Maximum Number of Vowels in a Substring of Given Length/code.cpp
class Solution {
public:
    int maxVowels(string s, int k) {
        int res = 0, cur = 0;
        for(int i = 0; i < k - 1 && i < s.size(); i ++)
            cur += is_vowel(s[i]);

        for(int i = k - 1; i < s.size(); i ++){
            cur += is_vowel(s[i]);
            res = max(res, cur);
            cur -= is_vowel(s[i - (k - 1)]);
        }
        return res;
    }

private:
    bool is_vowel(char c){
        return c == 'a' || c == 'e' || c == 'i' || c == 'o' || c == 'u'; 
    }
};
